packages/providers/src/util.ts

```diff
@@ -98,23 +98,87 @@ export const calculateTransactionFee = ({
 
 const DEFAULT_BLOCK_EXPLORER_URL = 'https://fuellabs.github.io/block-explorer-v2';
 
+const getPathFromInput = (key: string, value: string | number | undefined) => {
+  switch (key) {
+    case 'address':
+      return `address/${value}`;
+    case 'txId':
+      return `transaction/${value}`;
+    case 'blockNumber':
+      return `block/${value}`;
+    default:
+      return `/${key}/${value}`;
+  }
+};
+
 /**
  * Builds a block explorer url based on and the given path, block explorer URL and provider URL
  */
 export const buildBlockExplorerUrl = ({
   blockExplorerUrl,
   path,
   providerUrl,
+  address,
+  txId,
+  blockNumber,
 }: {
   blockExplorerUrl?: string;
-  path: string;
+  path?: string;
   providerUrl?: string;
+  address?: string;
+  txId?: string;
+  blockNumber?: number;
 }) => {
   const explorerUrl = blockExplorerUrl || DEFAULT_BLOCK_EXPLORER_URL;
 
+  // make sure that only ONE or none of the following is defined: address, txId, blockNumber
+  const customInputParams = [
+    {
+      key: 'address',
+      value: address,
+    },
+    {
+      key: 'txId',
+      value: txId,
+    },
+    {
+      key: 'blockNumber',
+      value: blockNumber,
+    },
+  ];
+  const definedValues = customInputParams
+    .filter((param) => param.value)
```

Comment:
Here I assume you want to filter all items with a non-null `value`. For this purpose, I think we could do it like:

```suggestion
    .filter((param) => !!param.value)
```